### DISCUSSION TITLE
Can't retrieve translation for pivot table column.

### DISCUSSION THREAD

**Discussion by smylmrz** [spatie/laravel-translatable#243] [Q&A]
I have many to many relation between product and attribute tables, and the pivot table has a value column which is translatable. ProductAttribute model is configured correctly, but for some reason, the value returns the JSON, not the translation. 
![1](https://user-images.githubusercontent.com/58288990/100548447-1092e180-3286-11eb-8e25-22270e4519d9.jpg)
![2](https://user-images.githubusercontent.com/58288990/100548451-12f53b80-3286-11eb-812f-9f549bec0c03.jpg)

**Comment by zjbarg** [comment]
How are you storing and retrieving `value` ?

**Comment by smylmrz** [comment]
@ZaidBarghouthi the pivot table has `product_id`, `attribute_id` and `value` columns. `Value` column is `$translatable`. I'm retrieving it with `$product->pivot->value`. I get the value in json as you can see.

**Comment by smylmrz** [comment]
Product and Attribute have a many-to-many relation. I found a workaround. So instead of calling it as a pivot column, I made a new relation with pivot and product table with one-to-many relation, which I don't like but seems working. For some reason, pivot columns are not translated.

**Comment by rkaalma** [comment]
I was troubleshooting a similar issue and found this discussion. I was using [Many to Many (Polymorphic)](https://laravel.com/docs/11.x/eloquent-relationships#many-to-many-polymorphic-relations) relationship with Spatie Translatable pivot column. DB structure something similar:

```
posts
    id - integer
    name - string
 
videos
    id - integer
    name - string
 
tags
    id - integer
    name - string
 
taggables
    tag_id - integer
    taggable_id - integer
    taggable_type - string
    some_text - json
```

I created a model for my "Taggables":

```
# app/models/Taggable.php`

...

class Taggable extends MorphPivot
{
    use HasTranslations;

    protected $table = 'taggables';

    protected $fillable = ['tag_id', 'taggable_id', 'taggable_type', 'some_text'];

    public $translatable = ['some_text'];
}
```

```
# app/Models/Post.php

...

class Post extends Model
{
    ...

    public tags(): MorphToMany
    {
        return $this->morphToMany(Tag::class, 'taggable')->using(Taggable::class)->withPivot('some_text');
    {
{
```
